internal/storage/device_session.go

```diff
@@ -486,12 +486,12 @@ func GetDeviceSessionForPHYPayload(p *redis.Pool, phy lorawan.PHYPayload, txDR,
 		}
 
 		// the FCnt is valid, validate the MIC
-		macPL.FHDR.FCnt = fullFCnt
```

Comment:
By removing this line, the `ValidateUplinkDataMIC` would only validate against the 16LSB, wouldn't it? The MIC validation needs the full 32bit frame-counter for validation.

Reply:
the description is issue#412
otherwise，ValidateAndGetFullFCntUp()  allow a gap， so macPL.FHDR.FCnt != fullFcnt is expected sometimes，if do not move that line， how can we pass the validate?

Reply:
sorry， my mistake，  The MIC validation needs the full 32bit frame-counter for validation
i will make a new pr